csrc/rdrand.cpp

```diff
@@ -268,13 +278,30 @@ bool rdseed_fallback(uint64_t *dest) {
 
 bool rng_retry_rdrand(uint64_t *dest) {
     int tries = 10;
-
+        uint64_t oldValue = 13; // Must not be initialized to 0 or UINT64_MAX. Anything else is fine.
     do {
-        if (likely(rng_rdrand(dest))) {
-            return true;
-        }
-
-        pause_and_decrement(tries);
+      if (likely(rng_rdrand(dest))) {
+        // Some AMD CPUs will find that RDRAND "sticks" on all 1s but still reports success.
+        // If we encounter this suspicious value (a 1/2^64 chance) we'll generate a second
```

Comment:
Technically, since we're also checking for 0, this is a 1/2^63 chance.

Reply:
I'll fix the comment.